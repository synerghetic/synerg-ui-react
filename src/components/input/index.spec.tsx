import React from 'react'
import Input from '.'
import { mount } from 'enzyme'

describe('Input', () => {
    it('should render', () => {
        const component = mount(<Input onChange={() => void 0} />)

        expect(component.html()).toContain('input')
        expect(component.html()).toMatchSnapshot()
    })

    it('should call onChange and update the variable', () => {
        let value = ''
        const onChange = (newValue: string) => {
            value = newValue
        }

        const component = mount(<Input onChange={onChange} />)

        const changeValue = 'Hello World'

        component
            .find('input')
            .simulate('change', { target: { value: changeValue } })

        expect(value).toBe(changeValue)
    })

    it('should call onChange and update the variable', () => {
        const changeValue = 'Hello World'
        let value = ''
        const onEnterPressed = () => {
            value = changeValue
        }

        const component = mount(
            <Input onChange={() => {}} onEnterPressed={onEnterPressed} />,
        )

        component.find('input').simulate('keydown', { key: 'Enter' })

        expect(value).toBe(changeValue)
    })

    it('should have an error', () => {
        const error = 'Oh no D:'

        const component = mount(<Input onChange={() => void 0} error={error} />)

        const html = component.html()

        expect(html).toMatchSnapshot()

        expect(component).toContainMatchingElement('.error-message')
        expect(component).toContainMatchingElement('.input-icon--error')
        expect(html).toContain(error)
    })
})
