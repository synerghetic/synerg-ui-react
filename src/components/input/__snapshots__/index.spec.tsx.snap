// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should have an error 1`] = `"<div class=\\"flex:column\\"><div class=\\"relative\\"><input class=\\"input--error\\" type=\\"text\\" value=\\"\\"><svg width=\\"24\\" height=\\"24\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" fill=\\"#000\\" class=\\"input-icon--error\\"><path d=\\"M13.41 12l4.3-4.29a1 1 0 1 0-1.42-1.42L12 10.59l-4.29-4.3a1 1 0 0 0-1.42 1.42l4.3 4.29-4.3 4.29a1 1 0 0 0 0 1.42 1 1 0 0 0 1.42 0l4.29-4.3 4.29 4.3a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42z\\"></path></svg></div><span class=\\"error-message \\">Oh no D:</span></div>"`;

exports[`Input should render 1`] = `"<div class=\\"flex:column\\"><div class=\\"relative\\"><input class=\\"input\\" type=\\"text\\" value=\\"\\"></div><span class=\\"error-message hidden\\"></span></div>"`;
